wikipedia-dark.user.css

```diff
@@ -520,13 +620,22 @@
   .oo-ui-iconElement-icon:not(.mw-echo-notifications-badge), .RTsettings,
   #gtx-host, .jfk-bubble-closebtn,
   #pt-notifications-notice .mw-echo-notifications-badge:before,
-  .oo-ui-pendingElement-pending,
   #pt-notifications-alert .mw-echo-notifications-badge:before,
   img.mwe-math-fallback-image-display, #cnotice-toggle-icon {
     -webkit-filter: invert(100%) !important;
             filter: invert(100%) !important;
     background-color: transparent !important;
   }
+  .oo-ui-pendingElement-pending {
+    background-color: #ddd !important;
+    filter: invert(100%) !important;
```

Comment:
Do we need to use this sort of solution?

Reply:
Well, to the best of my knowledge, the only way to make the animation on `oo-ui-pendingElement-pending` dark is to invert it. The problem is, it sometimes has child elements that need to be un-inverted (the echo notification icons, and the elements on the loading banner in the VisualEditor).

These child elements also clash with the generic icon inversion rules in some cases, and `filter: none !important` isn't helping either. If you have a better idea (besides reorganizing all those generic selectors causing trouble), I'm open to suggestions.

Reply:
I need specific page links to look at what the problem is and work on solutions. But doing this is the last solution youde go for usually.

Reply:
The `oo-ui-pendingElement-pending` class adds a "barbershop" stripe loading animation to an OOUI header.

You can see it in action when you click on the notifications box while logged in (as the notifications load):

  - [Capture 1](https://i.imgur.com/D9D28oG.mp4)

You can also see it briefly on the save dialog after making an edit, then clicking the preview button:

  - [Capture 2](https://i.imgur.com/VqxWP1M.mp4)

I added the class manually in these screenshots, but the real animation would happen for a second or two.

Reply:
Here is the vendor style for that class:
```css
.oo-ui-pendingElement-pending {
  background-color: #eaecf0;
  background-image: -webkit-linear-gradient(135deg,#fff 25%,transparent 25%,transparent 50%,#fff 50%,#fff 75%,transparent 75%,transparent);
  background-image: -moz-linear-gradient(135deg,#fff 25%,transparent 25%,transparent 50%,#fff 50%,#fff 75%,transparent 75%,transparent);
  background-image: linear-gradient(135deg,#fff 25%,transparent 25%,transparent 50%,#fff 50%,#fff 75%,transparent 75%,transparent);
  background-size: 1.42857143em 1.42857143em;
  -webkit-animation: oo-ui-pendingElement-stripes 650ms linear infinite;
     -moz-animation: oo-ui-pendingElement-stripes 650ms linear infinite;
          animation: oo-ui-pendingElement-stripes 650ms linear infinite;
}
```
I guess we just need to invert the linear-gradient then. Lemme try something...

Reply:
Oh snap! I got it. :smiley: Here's what I ended up with:
```css
  .oo-ui-pendingElement-pending {
    background-color: var(--gray-1);
    background-image: -webkit-linear-gradient(135deg, black 25%, transparent 25%, transparent 50%, black 50%, black 75%, transparent 75%, transparent);
    background-image: -moz-linear-gradient(135deg, black 25%, transparent 25%, transparent 50%, black 50%, black 75%, transparent 75%, transparent);
    background-image: linear-gradient(135deg, black 25%, transparent 25%, transparent 50%, black 50%, black 75%, transparent 75%, transparent);
    background-size: 1.42857143em 1.42857143em;
    -webkit-animation: oo-ui-pendingElement-stripes 650ms linear infinite;
       -moz-animation: oo-ui-pendingElement-stripes 650ms linear infinite;
            animation: oo-ui-pendingElement-stripes 650ms linear infinite;
  }
```

Reply:
In future please just drag screenshots into the reply area here instead of linking them from imgur, I rather not have to click and go elsewhere to see something related.

Reply:
A screenshot isn't the best medium for capturing an animation though. I couldn't figure out how to make them show as a GIF instead of a video, so it wouldn't embed. I could just upload the video files here directly, I guess.